src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![feature(conservative_impl_trait)]
```

Comment:
I want to keep this crate on stable, but adding the async stuff behind a Cargo feature would be fine. Once `conservative_impl_trait` lands on stable, we can enable async by default.

Reply:
I want it on `stable` as well and that's why I kept the commit doing this bit separate. The second commit is preferred, but the higher barrier is something I'd like to avoid if possible.

Reply:
So what do you suggest? Only merge the first three commits and leave the rest for later? Merge everything, but hide it behind a Cargo feature? I think both would be find with me.

Reply:
How about we use the base 3 commits for now (since it's still pre-1.0, this isn't that much of a big deal). Note that one drawback of the base 3 commits is that it's a busy loop (it always schedules itself again), but the ability to be async is worth it to the projects I'm working on at least.

I'll open a PR with the fourth commit which can be used once `impl Trait` is available. I suppose the better API could be put behind a feature flag. Since the signature is different, do we want the old name to be `event_stream_stable` and reserve `event_stream` for the more idealistic implementation?

Reply:
Okay, sounds good. Let's just merge the first three.

> Note that one drawback of the base 3 commits is that it's a busy loop

It's probably best to clearly document that on `event_stream`.

> I'll open a PR with the fourth commit which can be used once `impl Trait` is available. I suppose the better API could be put behind a feature flag. Since the signature is different, do we want the old name to be `event_stream_stable` and reserve `event_stream` for the more idealistic implementation?

I don't don't quite follow. If we go ahead as you suggested, we won't need any features at all, because we'll never merge anything that's not stable. Or am I misunderstanding something?

Reply:
The API is different. I'd expect the base one to be deprecated once the preferable one is available.

Reply:
This PR now contains just the first three commits and I've updated the docstring as well. The name is still `event_stream`.

Reply:
Ah, I realize now that we were talking about different things. You were talking about the name of the method, while I thought we were still talking about Cargo features.

The current name, `event_stream`, is fine. When we merge your follow-up PR, we'll just change that method. This is still a `0.x` crate, after all.